main.cpp

```diff
@@ -46,5 +47,7 @@
 int
 main(const int argc, const char* const* const argv)
 {
+    freebsd::main(argc, argv);
+
```

Comment:
This explicitly ties this code to FreeBSD instead of abstracting out the implementation, which breaks other OS support (making the freebsd org fork unusable on other OS platforms).

Reply:
There is a global registry of execution environment managers, and they should be registered at the very beginning. It's kind of a plugin initialization time. FreeBSD related code is based on conditional build and for non-FreeBSD systems or for WITHOUT_JAIL FreeBSD instances it provides a stub which is involved into requirements check process (`execenv` conf variable), provides the flag that jail is not supported, and politely responds with a message why a jail based test is skipped.

As long as there is no any plugin-like system introduced yet I decided to make it very simple and explicit, not to have any magic (possibly fragile) behind FreeBSD related code initialization. What do you think?